nifi-mock/src/main/java/org/apache/nifi/util/MockProcessSession.java

```diff
@@ -756,6 +756,13 @@ public void transfer(FlowFile flowFile) {
             throw new IllegalArgumentException("I only accept MockFlowFile");
         }
 
+        // if the flowfile provided was created in this session (i.e. it's in currentVersions),
+        // then throw an exception indicating that you can't transfer flowfiles back to self.
+        // this mimics the behavior of StandardProcessSession
+        if(currentVersions.get(flowFile.getId()) != null) {
```

Comment:
This should only fire the exception if the `flowFile.getId()` is in `currentVersions` but not in `originalVersions` since those are the newly created FlowFiles.  If it's in both it's a queued file being transferred back to the queue.  You can see the problem if you run the `TestUpdateAttribute` tests.